MTA-2850: Installing and Configuring OCP Local (#4)
migtools/mta-documentation: docs/topics/mta-7-installing-web-console-on-ocp-local.adoc

// Module included in the following assemblies:
//
// * docs/web-console-guide/master.adoc

:_content-type: PROCEDURE
[id="mta-7-installing-web-console-on-ocp-local_{context}"]
= Installing and configuring the {ProductName} Operator in a Red Hat OpenShift Local environment

Red Hat OpenShift Local provides a quick and easy way to set up a local OpenShift cluster on your desktop or laptop. This allows you to test your applications and configuration parameters before sending them to production.

[id="mta-7-ocp-os-sys-reqs_{context}"]
== Operating system requirements

Red Hat OpenShift Local requires the following minimum version of a supported operating system:

> .Requirements on Microsoft Windows

Author: ```suggestion
=== Requirements on Microsoft Windows
```

> === Requirements on Microsoft Windows

Author: ```suggestion
.Requirements on Microsoft Windows
```

On Microsoft Windows, Red Hat OpenShift Local requires the Windows 10 Fall Creators Update (version 1709) or later. Red Hat OpenShift Local does not work on earlier versions of Microsoft Windows. Microsoft Windows 10 Home Edition is not supported.

> === Requirements on macOS

Author: ```suggestion
.Requirements on macOS
```

> .Requirements on macOS

Author: ```suggestion
=== Requirements on macOS
```

On macOS, Red Hat OpenShift Local requires macOS 11 Big Sur or later. Red Hat OpenShift Local does not work on earlier versions of macOS.

> .Requirements on Linux

Author: ```suggestion
=== Requirements on Linux
```

> === Requirements on Linux

Author: ```suggestion
.Requirements on Linux
```

On Linux, Red Hat OpenShift Local is supported only on the latest two Red Hat Enterprise Linux/CentOS 8 and 9 minor releases and on the latest two stable Fedora releases.

When using Red Hat Enterprise Linux, the machine running Red Hat OpenShift Local must be registered with the Red Hat Customer Portal.

Ubuntu 18.04 LTS or later and Debian 10 or later are not supported and might require manual set up of the host machine.
See Required software packages to install the required packages for your Linux distribution.

==== Required software packages for Linux

Red Hat OpenShift Local requires the `libvirt` and `NetworkManager` packages to run on Linux:

* *Fedora/Red Hat Enterprise Linux* issue:

+
[source,terminal]
----
sudo dnf install NetworkManage
----

* *Debian/Ubuntu* issue:
+
[source,terminal]
----
sudo apt install qemu-kvm libvirt-daemon libvirt-daemon-system network-manager
----

> === Installing the {ProductName} Operator in a Red Hat OpenShift Local environment

Author: ```suggestion
[id="mta-7-installing-ocp-local_{context}"]
== Installing the {ProductName} Operator in a Red Hat OpenShift Local environment
```

.. Download **pull secret**.

. Assuming you saved the archive in the `~/Downloads directory` follow these steps:

+
[source,terminal]
----
cd ~/Downloads
----

+
[source,terminal]
----
tar xvf crc-linux-amd64.tar.xz
----

. Copy the `crc` executable to it:

+
[source,terminal]
----
cp ~/Downloads/crc-linux-<version-number>-amd64/crc ~/bin/crc
----
. Add the `~/bin/crc` directory to your `$PATH` variable:

+
[source,terminal]
----
export PATH=$PATH:$HOME/bin/crc
----
+
[source,terminal]
----
echo 'export PATH=$PATH:$HOME/bin/crc' >> ~/.bashrc
----

. Disable telemetry, issue the following command:

+
[source,terminal]
----
crc config set consent-telemetry no
----

[NOTE]
====
For macOS, download the relevant *crc-macos-installer.pkg*.

. Navigate to *Downloads* using *Finder*.
. Double-click on `crc-macos-installer.pkg`.
====

[id="mta-7-set-up-ocp-local_{context}"]
== Setting up Red Hat OpenShift Local

The `crc setup` command performs operations to set up the environment of your host machine for the OpenShift Local instance.

The `crc setup` command creates the `~/.crc directory`

1. Set up your host machine for OpenShift Local:

+
[source,terminal]
----
crc setup
----

> 

Author: ```suggestion

[id="mta-7-starting-ocp-local_{context}"]
== Starting the instance

Red Hat OpenShift Local presets represent a managed container runtime, and the lower bounds of system resources required by the instance to run it. 

[NOTE]
==== 
* On Linux or macOS, ensure that your user account has permission to use the `sudo` command. 
* On Microsoft Windows, ensure that your user account can elevate to Administrator privileges.
====

The `crc start` command starts the Red Hat OpenShift Local instance and configured container runtime. It offers the following flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value

| -b, --bundle 
| string 
| Bundle path/URI - absolute or local path, http, https or docker URI (string, like 'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1' default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) (default "/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle") |   

|-c, –cpus 
|int 
|Number of CPU cores to allocate to the instance 
| 4

|–disable-update-check 
| 
|Do not check for update 
|

|-d, –disk-size 
|uint 
|Total size in GiB of the disk used by the instance 
|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 
|log level for example, `+debug+`, `+info+`, `+warn+`, `+error+` |`+info+`

|===

```

|-c, –cpus 
|int 
|Number of CPU cores to allocate to the instance 
| 4

|–disable-update-check 
| 
|Do not check for update 
|

|-d, –disk-size 
|uint 
|Total size in GiB of the disk used by the instance 
|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 
|log level for example, `+debug+`, `+info+`, `+warn+`, `+error+` |`+info+`

|===

> 

Author: ```suggestion

The default configuration creates a virtual machine (VM) with 4 virtual CPUs and 10GB of RAM. 

To increase the number of virtual CPUs to 6 and the memory to 20GB, run crc config like this:

+
[source,terminal]
----
crc config set cpus 6
----

+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, issue:

+
[source,terminal]
----
crc config view
----

.Example Output

+
[source,terminal]
----
- consent-telemetry               : yes
- cpus                                       : 6
- memory                                : 16384
----

```

// Module included in the following assemblies:
//
// * docs/web-console-guide/master.adoc

:_content-type: PROCEDURE
[id="mta-7-installing-web-console-on-ocp-local_{context}"]
= Installing and configuring the {ProductName} Operator in a {rh-ocp-local} environment

> {rh-ocp-local} provides a quick and easy way to set up a local OpenShift cluster on your desktop or laptop. This allows you to test your applications and configuration parameters before sending them to production.

Author: ```suggestion
{rh-ocp-local} provides a quick and easy way to set up a local OpenShift cluster on your desktop or laptop. This local cluster allows you to test your applications and configuration parameters before sending them to production.
```

| default `'/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle'`

|-c, –cpus
|int
|Number of CPU cores to assign to the instance
| 4

|–disable-update-check
|
|Do not check for update
|

|-d, –disk-size
|uint
|Total size in GiB of the disk used by the instance
|31

|-h, –help
|
|Help for start
|

|-m, –memory
|int
|MiB of memory to assign to the instance
|10752

|-n, –nameserver
| string |IPv4 address of name server to use for the instance
|

|-o, –output
|string
|Output format in JSON
|

|-p, –pull-secret-file
| string
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags
|Type
|Description
|Default value
|–log-level
|string
|log level for example:

* `+debug+`

* `+info+`

* `+warn+`

* `+error+`
|`+info+`

|===

The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10 GiB of RAM.

To increase the number of virtual CPUs to 6, the disk-size to 200 GiB, and the memory to 20 GiB, run crc config as follows:
+
[source,terminal]
----
crc config set cpus 6
----

> +

Author: ```suggestion

+
[source,terminal]
----
crc config set disk-size 200 
----

+
```

|===

|Flags 
|Type 
|Description 
|Default value

| -b, --bundle 
| string 
| Bundle path/URI - absolute or local path, http, https or docker URI (string, like 'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1' default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) (default "/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle") |   

|-c, –cpus 
|int 
|Number of CPU cores to allocate to the instance 
| 4

|–disable-update-check 
| 
|Do not check for update 
|

|-d, –disk-size 
|uint 
|Total size in GiB of the disk used by the instance 
|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 

> |log level for example, `+debug+`, `+info+`, `+warn+`, `+error+` |`+info+`

Author: ```suggestion
|log level for example:
* `+debug+`
* `+info+`
* `+warn+` 
* `+error+` 
|`+info+`
```

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value

| -b, --bundle 
| string 
| Bundle path/URI - absolute or local path, http, https or docker URI (string, like 'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1' default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) (default "/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle") |   

|-c, –cpus 
|int 
|Number of CPU cores to allocate to the instance 
| 4

|–disable-update-check 
| 
|Do not check for update 
|

|-d, –disk-size 
|uint 
|Total size in GiB of the disk used by the instance 
|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

> [width="100%",cols="17%,15%,34%,34%",options="header",]
> |===
> 
> |Flags 
> |Type 
> |Description 
> |Default value
> |–log-level 
> |string 
> |log level for example:
> * `+debug+`
> * `+info+`
> * `+warn+` 
> * `+error+` 
> |`+info+`
> 
> |===

Author: ```suggestion
[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 
|log level for example:

* `+debug+`

* `+info+`

* `+warn+` 

* `+error+` 
|`+info+`

|===
```

// Module included in the following assemblies:
//
// * docs/web-console-guide/master.adoc

:_content-type: PROCEDURE
[id="mta-7-installing-web-console-on-ocp-local_{context}"]
= Installing and configuring the {ProductName} Operator in a {rh-ocp-local} environment

> {rh-ocp-local} provides a quick and easy way to set up a local OpenShift cluster on your desktop or laptop. This local cluster allows you to test your applications and configuration parameters before sending them to production.

Reviewer: We don't have an attribute for "OpenShift" in the MTA repo, we might want to make one.

[id="mta-7-ocp-os-sys-reqs_{context}"]
== Operating system requirements

{rh-ocp-local} requires the following minimum version of a supported operating system:

[id="mta-7-ocp-local-windows-reqs_{context}"]
=== {rh-ocp-local} requirements on Microsoft Windows

On Microsoft Windows, {rh-ocp-local} requires the Windows 10 Fall Creators Update (version 1709) or later. {rh-ocp-local} does not run on earlier versions of Microsoft Windows. Microsoft Windows 10 Home Edition is not supported.

[id="mta-7-ocp-local-macos-reqs_{context}"]
=== {rh-ocp-local} requirements on macOS

On macOS, {rh-ocp-local} requires macOS 11 Big Sur or later. {rh-ocp-local} does not run on earlier versions of macOS.

[id="mta-7-ocp-local-local-reqs_{context}"]
=== {rh-ocp-local} requirements on Linux

On Linux, {rh-ocp-local} is supported only on the latest two Red Hat Enterprise Linux 8 and 9 minor releases and on the latest two stable Fedora releases.

When using Red Hat Enterprise Linux, the machine running {rh-ocp-local} must be registered with the Red Hat Customer Portal.

Ubuntu 18.04 LTS or later and Debian 10 or later are not supported and might require manual setup of the host machine.

> 

Reviewer: Were earlier versions of Ubuntu and Debian supported?

==== Required software packages for Linux

{rh-ocp-local} requires the `libvirt` and `NetworkManager` packages to run on Linux:

* On *Fedora* and *Red Hat Enterprise Linux* run:

+
[source,terminal]
----
sudo dnf install NetworkManager
----

> * *Debian/Ubuntu* run:

Reviewer: Suggestion: "On *Debian/Ubuntu* run...

> * *Debian/Ubuntu* run:

Author: ```suggestion
* On *Debian/Ubuntu* run:
```

+
[source,terminal]
----
sudo apt install qemu-kvm libvirt-daemon libvirt-daemon-system network-manager
----

[id="mta-7-installing-ocp-local_{context}"]
== Installing the {ProductName} Operator in a {rh-ocp-local} environment

To install OpenShift Local:

> 

Reviewer: Maybe "To install {rh-ocp-local}:"  ?

Author: Eyes like a damn hawk! LOL

> To install OpenShift Local:

Author: ```suggestion
To install {rh-ocp-local}:
```

. Download the latest release of link:https://console.redhat.com/openshift/create/local[{rh-ocp-local}]( for your platform.

> 

Reviewer: Extra ( before "for your platform".

> . Download the latest release of link:https://console.redhat.com/openshift/create/local[{rh-ocp-local}]( for your platform.

Author: ```suggestion
. Download the latest release of link:https://console.redhat.com/openshift/create/local[{rh-ocp-local}] for your platform.
```

.. Download *OpenShift Local*.

.. Download *pull secret*.

> . Assuming you saved the archive in the `~/Downloads directory` follow these steps:

Author: ```suggestion
. Assuming you saved the archive in the `~/Downloads directory`, follow these steps:
```

> 

Reviewer: Suggestion: "...  `~/Downloads directory`, ..."

+
[source,terminal]
----
cd ~/Downloads
----

+
[source,terminal]
----
tar xvf crc-linux-amd64.tar.xz
----

. Copy the `crc` executable to it:

+
[source,terminal]
----
cp ~/Downloads/crc-linux-<version-number>-amd64/crc ~/bin/crc
----
. Add the `~/bin/crc` directory to your `$PATH` variable:

+
[source,terminal]
----
export PATH=$PATH:$HOME/bin/crc
----
+
[source,terminal]
----
echo 'export PATH=$PATH:$HOME/bin/crc' >> ~/.bashrc
----

. To disable telemetry, run the following command:

+
[source,terminal]
----
crc config set consent-telemetry no
----

[NOTE]
====
For macOS, download the relevant *crc-macos-installer.pkg*.

. Navigate to *Downloads* using *Finder*.
. Double-click on `crc-macos-installer.pkg`.
====

[id="mta-7-set-up-ocp-local_{context}"]
== Setting up {rh-ocp-local}

The `crc setup` command performs operations to set up the environment of your host machine for the {rh-ocp-local} instance.

> The `crc setup` command creates the `~/.crc directory`

Reviewer: Line 120 needs a period.

> The `crc setup` command creates the `~/.crc directory`

Author: ```suggestion
The `crc setup` command creates the `~/.crc directory`.
```

1. Set up your host machine for OpenShift Local:

> 

Reviewer: Line 122:  Suggestions: Remove "1." and use the attribute for "OpenShift local".

> 1. Set up your host machine for OpenShift Local:

Author: ```suggestion
. Set up your host machine for  {rh-ocp-local}:
```

> . Set up your host machine for  {rh-ocp-local}:

Author: ```suggestion
. Set up your host machine for {rh-ocp-local}:
```

|===

|Flags
|Type
|Description
|Default value

| -b, --bundle
| string
| Bundle path/URI - absolute or local path, HTTP, HTTPS or docker URI, for example, `'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1'`
| default `'/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle'`

|-c, –cpus
|int
|Number of CPU cores to assign to the instance
| 4

|–disable-update-check
|
|Do not check for update
|

|-d, –disk-size
|uint
|Total size in GiB of the disk used by the instance
|31

|-h, –help
|
|Help for start
|

|-m, –memory
|int
|MiB of memory to assign to the instance
|10752

|-n, –nameserver
| string |IPv4 address of name server to use for the instance
|

|-o, –output
|string
|Output format in JSON
|

|-p, –pull-secret-file
| string
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags
|Type
|Description
|Default value
|–log-level
|string
|log level for example:

* `+debug+`

* `+info+`

* `+warn+`

* `+error+`
|`+info+`

|===

> The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31GiB, and 10 GiB of RAM.

Author: ```suggestion
The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10 GiB of RAM.
```

> The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10 GiB of RAM.

Author: ```suggestion
The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10 GiB of RAM. However, this default configuration is not sufficent to stably run {ProductShortName}.
```

> 

Reviewer: Is there an extra space before "31"?

> The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10 GiB of RAM.

Author: ```suggestion
The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31GiB, and 10 GiB of RAM.
```

> To increase the number of virtual CPUs to 6, the disk-size to 200 GiB, and the memory to 20 GiB, run crc config as follows:

Reviewer: Suggestion: ...run `crc config`

> To increase the number of virtual CPUs to 6, the disk-size to 200 GiB, and the memory to 20 GiB, run crc config as follows:

Author: ```suggestion
To increase the number of virtual CPUs to 6, the disk-size to 200 GiB, and the memory to 20 GiB, run `crc config` as follows:
```

+
[source,terminal]
----
crc config set cpus 6
----

+
[source,terminal]
----
crc config set disk-size 200 
----

+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, run:
+
[source,terminal]
----
crc config view
----

.Example Output
+
[source,terminal]
----
- consent-telemetry    : yes
- cpus                 : 6
- disk-size            : 200
- memory               : 16384
----

[NOTE]
====
Changes to the inputted configuration property are only applied when the CRC instance is started.

> If you already have a running CRC instance, then for this configuration change to take effect, stop the CRC instance with `crc stop` and restart it with `crc start`.

Author: ```suggestion
If you already have a running CRC instance, for this configuration change to take effect, stop the CRC instance with `crc stop` and restart it with `crc start`.
```

For macOS, download the relevant *crc-macos-installer.pkg*.

. Navigate to *Downloads* using *Finder*.
. Double-click on `crc-macos-installer.pkg`.
====

[id="mta-7-set-up-ocp-local_{context}"]
== Setting up Red Hat OpenShift Local

The `crc setup` command performs operations to set up the environment of your host machine for the OpenShift Local instance.

The `crc setup` command creates the `~/.crc directory`

1. Set up your host machine for OpenShift Local:

+
[source,terminal]
----
crc setup
----

[id="mta-7-starting-ocp-local_{context}"]
== Starting the instance

Red Hat OpenShift Local presets represent a managed container runtime, and the lower bounds of system resources required by the instance to run it. 

[NOTE]
==== 
* On Linux or macOS, ensure that your user account has permission to use the `sudo` command. 
* On Microsoft Windows, ensure that your user account can elevate to Administrator privileges.
====

The `crc start` command starts the Red Hat OpenShift Local instance and configured container runtime. It offers the following flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value

| -b, --bundle 
| string 

> | Bundle path/URI - absolute or local path, http, https or docker URI (string, like 'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1' default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) (default "/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle") |   

Author: ```suggestion
| Bundle path/URI - absolute or local path, http, https or docker URI, for example, `'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1'` default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) 
| default `"/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle"`    
```

For macOS, download the relevant *crc-macos-installer.pkg*.

. Navigate to *Downloads* using *Finder*.
. Double-click on `crc-macos-installer.pkg`.
====

[id="mta-7-set-up-ocp-local_{context}"]
== Setting up Red Hat OpenShift Local

The `crc setup` command performs operations to set up the environment of your host machine for the OpenShift Local instance.

The `crc setup` command creates the `~/.crc directory`

1. Set up your host machine for OpenShift Local:

+
[source,terminal]
----
crc setup
----

[id="mta-7-starting-ocp-local_{context}"]
== Starting the instance

Red Hat OpenShift Local presets represent a managed container runtime, and the lower bounds of system resources required by the instance to run it. 

[NOTE]
==== 
* On Linux or macOS, ensure that your user account has permission to use the `sudo` command. 
* On Microsoft Windows, ensure that your user account can elevate to Administrator privileges.
====

The `crc start` command starts the Red Hat OpenShift Local instance and configured container runtime. It offers the following flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value

| -b, --bundle 
| string 

> | Bundle path/URI - absolute or local path, http, https or docker URI, for example, `'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1'` default '/home/<user>/.crc/cache/crc_libvirt_4.15.14_amd64.crcbundle' ) 

Author: ```suggestion
| Bundle path/URI - absolute or local path, http, https or docker URI, for example, `'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1'`  
```

> | default `"/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle"`    

Author: ```suggestion
| default `'/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle'`    
```

|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 
|log level for example:

* `+debug+`

* `+info+`

* `+warn+` 

* `+error+` 
|`+info+`

|===

The default configuration creates a virtual machine (VM) with 4 virtual CPUs and 10GB of RAM. 

To increase the number of virtual CPUs to 6 and the memory to 20GB, run crc config like this:
+
[source,terminal]
----
crc config set cpus 6
----
+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, issue:
+
[source,terminal]
----
crc config view
----

.Example Output
+
[source,terminal]
----
- consent-telemetry    : yes
- cpus                 : 6
- memory               : 16384
----

> 

Author: ```suggestion
[width="100%",cols="25%,25%,25%,25%",options="header",]
|===
|Memory (GiB) 
|CPU 
|Disk sze (GiB) 
|Comments

|15 
|4 (default)
|200 
|MTA does not run analysis due to insufficient memory and CPU.

|20 
|4 (default) 
|200 
|MTA does not run analysis due to insufficient CPU.

|17 
|5 
|50 
|The hub pod does not start due to disk size.

|17 
|5 
|200 
|The analysis is completed in approximately 3 minutes.

|20 
|5 
|100 
|The hub pod does not start due to disk size.

|20 
|5 
|110 
|The analysis is completed in approximately 5 minutes.

|20 
|5 
|150 
|The analysis is completed in under 3 minutes.

|20 
|5 
|200 
|The analysis is completed in approximately 2 minutes.

|20 
|5 
|35, with the MTA Operator configurations `+cache_data_volume_size+` and `+hub_bucket_volume_size+` set to 5GiB
|The analysis is completed in approximately 2 minutes
|===
```

|===

|Flags
|Type
|Description
|Default value

| -b, --bundle
| string
| Bundle path/URI - absolute or local path, http, https or docker URI, for example, `'https://foo.com/crc_libvirt_4.15.14_amd64.crcbundle', 'docker://quay.io/myorg/crc_libvirt_4.15.14_amd64.crcbundle:2.37.1'`
| default `'/home/<user>/.crc/cache/ crc_libvirt_4.15.14_amd64.crcbundle'`

|-c, –cpus
|int
|Number of CPU cores to allocate to the instance
| 4

|–disable-update-check
|
|Do not check for update
|

|-d, –disk-size
|uint
|Total size in GiB of the disk used by the instance
|31

|-h, –help
|
|Help for start
|

|-m, –memory
|int
|MiB of memory to allocate to the instance
|10752

|-n, –nameserver
| string |IPv4 address of nameserver to use for the instance
|

|-o, –output
|string
|Output format in JSON
|

|-p, –pull-secret-file
| string
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags
|Type
|Description
|Default value
|–log-level
|string
|log level for example:

* `+debug+`

* `+info+`

* `+warn+`

* `+error+`
|`+info+`

|===

> The default configuration creates a virtual machine (VM) with 4 virtual CPUs and 10GB of RAM.

Author: ```suggestion
The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10GiB of RAM.
```

> To increase the number of virtual CPUs to 6 and the memory to 20GB, run crc config like this:

Author: ```suggestion
To increase the number of virtual CPUs to 6, the disk-size to 200 GiB and the memory to 20GiB, run crc config like this:
```

+
[source,terminal]
----
crc config set cpus 6
----
+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, run:
+
[source,terminal]
----
crc config view
----

.Example Output
+
[source,terminal]
----
- consent-telemetry    : yes

> - cpus                 : 6

Author: ```suggestion
- cpus                 : 6
- disk-size          : 200
```

|

|-m, –memory
|int
|MiB of memory to allocate to the instance
|10752

|-n, –nameserver
| string |IPv4 address of nameserver to use for the instance
|

|-o, –output
|string
|Output format in JSON
|

|-p, –pull-secret-file
| string
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags
|Type
|Description
|Default value
|–log-level
|string
|log level for example:

* `+debug+`

* `+info+`

* `+warn+`

* `+error+`
|`+info+`

|===

The default configuration creates a virtual machine (VM) with 4 virtual CPUs, a disk size of  31 GiB, and 10GiB of RAM.

To increase the number of virtual CPUs to 6, the disk-size to 200 GiB, and the memory to 20GiB, run crc config as follows:
+
[source,terminal]
----
crc config set cpus 6
----
+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, run:
+
[source,terminal]
----
crc config view
----

.Example Output
+
[source,terminal]
----
- consent-telemetry    : yes
- cpus                 : 6
- disk-size          : 200
- memory               : 16384
----

> 

Author: ```suggestion

[NOTE] 
====
Changes to the inputted configuration property are only applied when the CRC instance is started.

If you already have a running CRC instance, then for this configuration change to take effect, stop the CRC instance with `crc stop` and restart it with `crc start`.
====

```

|31

|-h, –help 
| 
|Help for start 
|

|-m, –memory 
|int 
|MiB of memory to allocate to the instance 
|10752

|-n, –nameserver 
| string |IPv4 address of nameserver to use for the instance 
|

|-o, –output 
|string 
|Output format in JSON 
|

|-p, –pull-secret-file 
| string 
|File path of image pull secret (download from https://console.redhat.com/openshift/create/local) |
|===

It also offers the following global flags:

[width="100%",cols="17%,15%,34%,34%",options="header",]
|===

|Flags 
|Type 
|Description 
|Default value
|–log-level 
|string 
|log level for example:

* `+debug+`

* `+info+`

* `+warn+` 

* `+error+` 
|`+info+`

|===

The default configuration creates a virtual machine (VM) with 4 virtual CPUs and 10GB of RAM. 

To increase the number of virtual CPUs to 6 and the memory to 20GB, run crc config like this:
+
[source,terminal]
----
crc config set cpus 6
----
+
[source,terminal]
----
$ crc config set memory 20480
----

To check the configuration, issue:
+
[source,terminal]
----
crc config view
----

.Example Output
+
[source,terminal]
----
- consent-telemetry    : yes
- cpus                 : 6
- memory               : 16384
----

> 

Author: ```suggestion

The following table shows the different configurations of {rh-ocp-local} tested and the results obtained:
```
